### ISSUE TITLE
Can't escape `E:\$RECYCLE.BIN` exclude on Windows

### ISSUE THREAD

**Issue by Ambient-Impact** [creativeprojects/resticprofile#215]
Hi there. First off, thank you for this project because it seems to be really well thought out and incredibly useful, and I know how much work FOSS stuff can be.

I'm running into an issue on Windows where I'm trying to back up an entire local drive (`E:\`) which is just lots of files, so not operating system or running programs. I'd prefer to pass restic the drive root so that I don't have to remember to add any new folders in the drive to my back ups and just exclude a couple of operating system created directories to avoid errors. The one that's giving me problems is the recycle bin directory, named `E:\$RECYCLE.BIN` which I can't seem to escape or quote correctly in PowerShell no matter what I try. The backtick (`` ` ``) seems to have no effect. My workaround for now is just to manually list all the top level folders I want to include in the back up, but other wrappers such as [prestic](https://github.com/ducalex/prestic) seemed to handle this fine and I kind of miss that.

Current work in progress messy `profile.yaml`:

```yaml
# @see https://creativeprojects.github.io/resticprofile/configuration/examples/#configuration-example-for-windows

version: '1'

global:
#   shell: 'bash'
  priority: 'idle'
  restic-arguments-filter: false

default:
  repository: 'rclone:lan-server:backup_desktop/restic'
  password-file: 'C:\Users\i\restic\restic-lan-server-password.txt'
  initialize: false

  backup:

    verbose: true

    source:
      - E:\
      # - 'E:\Design'
      # - 'E:\Disk Images'
      # - 'E:\Files'
      # - 'E:\MonitoRSS back ups'
      # - 'E:\Repository'
      # - 'E:\Web'

    exclude:
      - 'E:\$RECYCLE.BIN'
    #   # - '"E:\$RECYCLE.BIN"'
      - 'E:\System Volume Information'

    # # files with no path are relative to the configuration file
    # exclude-file:
    #   - exclude.txt

    check-after: true

    # run-before: dir /l
    # run-after: echo All Done!

    # ignore restic warnings
    # without it the backup is considered failed when restic can't read some files
    no-error-on-warning: true

    # Every day at 11 AM.
    #
    # @todo Can this be set to run after the Acronis back up has completed
    #  via a post command in the Acronis options?
    schedule: '*-*-* 11:00:00'

    # @todo Use VSS snapshot; requires backup permissions.
    # @see https://restic.readthedocs.io/en/stable/040_backup.html
    #use-fs-snapshot: true

  # retention is a special section that run the 'forget' command
  # before or after a backup
  retention:
    before-backup: false
    after-backup: true

    # Adapted from Autorestic docs:
    # @see https://autorestic.vercel.app/location/forget

    # Always keep at least 5 snapshots.
    keep-last: 5
    # Keep 3 last hourly snapshots.
    keep-hourly: 3
    # Keep 4 last daily snapshots.
    keep-daily: 4
    # Keep 1 last weekly snapshots.
    keep-weekly: 1
    # Keep 12 last monthly snapshots.
    keep-monthly: 12
    # Keep 7 last yearly snapshots.
    keep-yearly: 7
    # Keep snapshots from the last 14 days.
    keep-within: 14d

    # keep-daily: 1
    # keep-hourly: 1
    # keep-weekly: 1
    # keep-monthly: 1
    # keep-within: 14d
    # keep-tag:
    #   - forever

    prune: true
    # prune: false
    # # tag can be a boolean ('true' meaning to copy tag set from 'backup')
    # # or a custom set of tags.
    # # Default is 'false', meaning that tags are NOT used.
    # tag: true

    # # host can be a boolean ('true' meaning current hostname)
    # # or a string to specify a different hostname
    # host: true
```

And here's the output of `resticprofile --dry-run backup`:

```powershell
PS C:\Users\i\restic> resticprofile --dry-run backup
2023/05/26 06:01:13 using configuration file: C:\Users\i\restic\resticprofile\profiles.yaml
2023/05/26 06:01:14 profile 'default': starting 'backup'
2023/05/26 06:01:14 dry-run: C:\Users\i\scoop\shims\restic.exe backup --exclude=E:\.BIN --exclude=E:\System Volume Information --password-file=C:\Users\i\restic\restic-lan-server-password.txt --repo=rclone:lan-server:backup_desktop/restic --verbose E:\
2023/05/26 06:01:14 profile 'default': finished 'backup'
2023/05/26 06:01:14 profile 'default': cleaning up repository using retention information
2023/05/26 06:01:14 dry-run: C:\Users\i\scoop\shims\restic.exe forget --keep-daily=4 --keep-hourly=3 --keep-last=5 --keep-monthly=12 --keep-weekly=1 --keep-within=14d --keep-yearly=7 --password-file=C:\Users\i\restic\restic-lan-server-password.txt --path=E:\ --prune --repo=rclone:lan-server:backup_desktop/restic
2023/05/26 06:01:14 profile 'default': checking repository consistency
2023/05/26 06:01:14 dry-run: C:\Users\i\scoop\shims\restic.exe check --password-file=C:\Users\i\restic\restic-lan-server-password.txt --repo=rclone:lan-server:backup_desktop/restic
```

**Comment by jkellerer** [comment]
Hi @Ambient-Impact, what you describe is a side effect of the environment variable resolution function in all path variables (that is platform neutral and uses $VAR or ${VAR} syntax).

Unfortunately, this clashes with windows $ paths and we should probably preserve `$VAR` if it doesn't resolve, provide escaping (e.g. `$$` for a single `$`) or even make it configurable if env variables are resolved at all. None of this exists at the moment.

As a workaround for the moment, how about:
```yaml
      exclude:
        - '*RECYCLE.BIN*'
```

**Comment by Ambient-Impact** [comment]
Awesome, that worked! Thanks! Agreed, some kind of escaping would be helpful, and `$$` makes sense.

**Comment by jkellerer** [comment]
Escaping with `$$` is included in `0.23`. Closing case.
